src/NGen.cpp

```diff
@@ -541,11 +541,14 @@ int main(int argc, char* argv[]) {
     // T-ROUTE data storage
     std::unordered_map<std::string, int> nexus_indexes;
 #if NGEN_WITH_ROUTING
-    size_t nexus_collection_size = nexus_collection->get_size();
-    for (int i = 0; i < nexus_collection_size; ++i) {
+    int nexus_index = 0;
+    for (int i = 0; i < nexus_collection->get_size(); ++i) {
         auto feature = nexus_collection->get_feature(i);
         std::string feature_id = feature->get_id();
-        nexus_indexes[feature_id] = i;
+        if (feature_id.compare(0, 4, "nex-") == 0) {
```

Comment:
This may need to be a call to `isNexus` from `HY_Features_IDs.hpp`, to ensure coverage of other nexus identifiers, like terminal `tnx-`, coastal `cnx-`, and inland `inx-`.

Out of curiosity, which features are intentionally being excluded by this check? Is it the terminal sentinels, or are catchments in the collection, or something else?